src/pool/solo/stratum_proxy.py

```diff
@@ -0,0 +1,210 @@
+        self.cli_queue.put(subscribe+'\n')
+
+    def serverDataReceived(self, chunk):
+        if chunk is False:
+            self.cli_queue = None
+            log.msg("Client: disconnecting from peer")
+            self.factory.continueTrying = False
+            self.transport.loseConnection()
+        elif self.cli_queue:
+            if verbose:
+                log.msg("Client: writing %d bytes to peer" % len(chunk))
+            self.transport.write(chunk)
+            self.cli_queue.get().addCallback(self.serverDataReceived)
+        else:
+            self.factory.cli_queue.put(chunk)
+
+    def dataReceived(self, chunk):
+        global extra_nonce
+        global verbose
+        if verbose:
+            log.msg("Client: %d bytes received from peer" % len(chunk))
+        curstr = chunk.splitlines()
+        disconnectflag = False
+        for val in curstr:
+            try:
+                data = fromJson(val)
+                if data.has_key('method'):
+                    if data.get('method') == 'mining.set_difficulty':
+                        self.cli_diff   = int(data.get('params')[0])
+                        log.msg("diff from stratum = %d" % self.cli_diff)
+                        if self.cli_diff < 1:    # it should not be happen but anyway
+                            self.cli_diff = 1
+                        self.cli_target = template.difficulty_to_hextarget(self.cli_diff)
+                        modifiedchunk   = "set_target %s diff %d" % (self.cli_target, self.cli_diff)
+                    elif data.get('method') == 'mining.notify':
+                        self.cli_wbclean = data.get('params')[8]
+                        if self.cli_wbclean and extra_nonce > 0:
+                            extra_nonce = 0
+                        else:
+                            extra_nonce += 1
+                        p_header = template.get_params_header(data.get('params'), self.cli_enonce1, extra_nonce, self.cli_n2len)
+                        self.cli_idstring = str(data.get('params')[0])
+                        self.cli_time     = str(data.get('params')[7])
+                        self.cli_odokey   = int(data.get('odokey'))
+                        self.cli_nonce2   = template.n2hex(extra_nonce, self.cli_n2len)
+                        modifiedchunk = "work %s %s %d %s %s %s" % (p_header, self.cli_target, self.cli_odokey, self.cli_idstring, self.cli_time, self.cli_nonce2)
+                    else:
+                        modifiedchunk = val   # send unmodified content
+                elif data.has_key('reject-reason'):
+                         if str(data.get('reject-reason')) == "Stale":
+                             modifiedchunk = "result stale"
+                         else:
+                             modifiedchunk = "result inconclusive"
+                elif data.has_key('result'):
+                    if data.get('result') == True:
+                         modifiedchunk = "result accepted"
```

Comment:
You seem to be sending a "result accepted" message immediately after authentication.